apps/prairielearn/src/components/TopicBadge.html.ts

```diff
@@ -0,0 +1,8 @@
+import { html } from '@prairielearn/html';
+
+import type { Topic } from '../lib/db-types.js';
+
+export function TopicBadge(topic: Topic | null) {
+  if (topic == null) return html`—`;
```

Comment:
When would we expect a topic to be null? Put differently, why wasn't this in the partial before?

Reply:
I originally did it because, in the assessment questions page, the schema had the topic as nullable, but I reviewed the query and the topic would never be null since the JOIN on topics is a regular inner join. I changed this to a required topic instead in 054e930.